v0/destinations/salesforce/utils.js

```diff
@@ -0,0 +1,154 @@
+const { handleHttpRequest } = require("../../../adapters/network");
+const { isHttpStatusSuccess } = require("../../util");
+const Cache = require("../../util/cache");
+const { TRANSFORMER_METRIC } = require("../../util/constant");
+const { ApiError } = require("../../util/errors");
+const {
+  ACCESS_TOKEN_CACHE_TTL,
+  SF_TOKEN_REQUEST_URL_SANDBOX,
+  SF_TOKEN_REQUEST_URL,
+  DESTINATION
+} = require("./config");
+
+const ACCESS_TOKEN_CACHE = new Cache(ACCESS_TOKEN_CACHE_TTL);
+
+// Utility method to construct the header to be used for SFDC API calls
+// The "Authorization: Bearer <token>" header element needs to be passed for
+// authentication for all SFDC REST API calls
+const getAccessToken = async destination => {
+  const accessTokenKey = destination.ID;
+
+  return ACCESS_TOKEN_CACHE.get(accessTokenKey, async () => {
+    let SF_TOKEN_URL;
+    if (destination.Config.sandbox) {
+      SF_TOKEN_URL = SF_TOKEN_REQUEST_URL_SANDBOX;
+    } else {
+      SF_TOKEN_URL = SF_TOKEN_REQUEST_URL;
+    }
+    const authUrl = `${SF_TOKEN_URL}?username=${
+      destination.Config.userName
+    }&password=${encodeURIComponent(
+      destination.Config.password
+    )}${encodeURIComponent(destination.Config.initialAccessToken)}&client_id=${
+      destination.Config.consumerKey
+    }&client_secret=${destination.Config.consumerSecret}&grant_type=password`;
+    const { httpResponse, processedResponse } = await handleHttpRequest(
+      "post",
+      authUrl,
+      {}
+    );
+    // If the request fails, throwing error.
```

Comment:
should we use `salesforceResponseHandler` here ?

Reply:
yes, updated, and also changed customError to TransformationError in transformer.